### ISSUE TITLE
Intro / Dependencies example / EIP does not work

### ISSUE THREAD

**Issue by lyrixx** [hashicorp/terraform#1666]
Hi.

I'm following the [quick start guide](https://www.terraform.io/intro/getting-started/dependencies.html), and I hit an error:

```
$ terraform apply
aws_instance.elasticsearch: Refreshing state... (ID: i-e6f5d400)
aws_eip.ip: Creating...
  allocation_id:  "" => "<computed>"
  association_id: "" => "<computed>"
  domain:         "" => "<computed>"
  instance:       "" => "i-e6f5d400"
  private_ip:     "" => "<computed>"
  public_ip:      "" => "<computed>"
aws_eip.ip: Error: 1 error(s) occurred:

* Error creating EIP: could not find credentials configuration.
Error applying plan:

1 error(s) occurred:

* 1 error(s) occurred:

* 1 error(s) occurred:

* Error creating EIP: could not find credentials configuration.

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

BTW, the error rendering is a bit ugly (There is many times `1 error(s) occurred`)

And this is my config file:

```
provider "aws" {
    access_key = "XXXXXXXXXXXXXXXXXXXXXXXXXxxx"
    secret_key = "XXXXXXXXXXXXXXXXXXXXXXXXXxxx"
    region = "eu-west-1"
}

resource "aws_instance" "elasticsearch" {
    ami = "ami-e92c419e"
    instance_type = "t1.micro"
}

resource "aws_eip" "ip" {
    instance = "${aws_instance.elasticsearch.id}"
}
```

**Comment by Mihailoff** [comment]
+1

**Comment by phinze** [comment]
Sorry for the bad experience here, and thanks for the reports.

Can you both confirm that you're using the latest released version of terraform - 0.4.2?

@Mihailoff AWS does set the default limit on EIPs pretty low, but it should allow 5 EIPs per region. The pastebin you linked has a panic in it though so I'm tagging this thread `crash` to make sure we take a look at that soon.

**Comment by lyrixx** [comment]
Wep, I confirm I'm using the last (0.4.2 version (linux amd64))

**Comment by lyrixx** [comment]
BTW, I just tested with master and it works well

```
terraform version
Terraform v0.5.0-dev (3d367794e93ce56fb7f6f29f2e5035ea62c6e0d3)
```

**Comment by Mihailoff** [comment]
Me as well `0.4.2`

**Comment by lyrixx** [comment]
BTW, the vpc and subnet does not work too

**Comment by catsby** [comment]
This issue is fixed in v0.5.0, sorry for the trouble here.

> BTW, the vpc and subnet does not work too

@lyrixx is this still happening in v0.5.0? Do you have a config that reproduces this (minus secrets)? If so, could you please open a new issue? Thanks!
